Refactor

jesford opened the pull request.

QuantumPlatypus commented on safespace/mgs/prefect_phenoscore.py:
```diff
@@ -261,7 +261,7 @@ def calculate_zfactor(*, relabeled_data,
             print(message_template.format(
                 zfactor_healthy_query, group_labels,
-                sub_wme[zfactor_healthy_query_column].unique()))
+                sub_wme[zfactor_healthy_query].unique()))
```
This should be zfactor_query_column.  Zfactor_healthy_query is a column string while zfactor_query_column is the name of the column that the zfactor_healthy_query uses.

jesford replied:
OK, well `zfactor_healthy_query_column` is not defined anywhere in the code, so I just took my best guess. Should it be an additional keyword argument that the function takes or something else?

jesford replied:
OK, I just took my best guess since that variable wasn't defined anywhere. `zfactor_healthy_query` is only used on this one line. It seems weird to pass a variable into a function who's only purpose is for a print statement that gets hit in one-off cases (I haven't had this print statement execute anyway). Is there a way this field could just be inferred from the other args or the data if its important to keep?

QuantumPlatypus commented on safespace/mgs/prefect_phenoscore.py:
```diff
@@ -208,72 +246,71 @@ def calculate_zfactor(*, relabeled_data,
     zfactor_same_healthy_groupby : list
-        list of column names from which to form groups for the zfactor
-        calculation.  Each group will have the same healthy population
+        Column names used to define zfactor groups. All groups will
```
This isn't quite right - Each group will share the same healthy population; different groups will have different healthy populations.

jesford commented on safespace/mgs/prefect_phenoscore.py:
```diff
@@ -70,136 +56,188 @@ def check_embeddings(experiment_labels, verbose=False):
-    """Uses metamorph to create a data bunch for the given experiment.
-    If modify_perts_func is provided, this function will be applied to the
-    perturbations before creating the metadata.
+def load_data_bunch(exp, job_id=None, modify_perts_func=None, fill_dict=None,
```
I'm open to this, but not too sure if its worth the effort. At the moment there is only 1 KR post that uses this code (and it will be updated shortly anyway). I don't really want to say 🤞 this will now be backwards compatibly since thats unlikely... What I have been doing is noting the git hash in my KR post in case we need to reference the right code later. What do you think about that solution? Its also error prone (like if we forget to update it), but its way less effort 🤷

jesford commented:
OK @QuantumPlatypus - I removed the confusing statement in the docstring. I also separated the file into submodules:
1. `data.py` contains all functions/tasks related to our 1st data-loading Flow.
2. `analysis.py` contains all functions/tasks related to the 2nd metrics/analysis Flow.
3. `plotting.py`
4. `utils.py`

I put these all in a subpackage called `fibrosis` (still under your `mgs/` dir) and added appropriates inits, so that you can do
`from safespace import fibrosis` and then access all the functions at that top level (i.e. bypass file names like `fibrosis.plot_heatmap`). I'm open to different names on any of the folders/files: `dataloader.py` instead of `data.py`?

QuantumPlatypus commented:
> Also, should `phenoscore_analysis.py` be removed? or are parts of this still useful and should go in the `fibrosis` subpackage?
I think it should be removed.
